### ISSUE TITLE
Add stubs for _async methods

### ISSUE THREAD

**Issue by dgilman-hrp** [auth0/auth0-python#615]
### Checklist

- [X] I have looked into the [Readme](https://github.com/auth0/auth0-python#readme) and [Examples](https://github.com/auth0/auth0-python/blob/master/EXAMPLES.md), and have not found a suitable solution or answer.
- [X] I have looked into the [API documentation](https://auth0-python.readthedocs.io/en/latest/) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/auth0-python/issues) and have not found a suitable solution or answer.
- [X] I have searched the [Auth0 Community](https://community.auth0.com) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Describe the problem you'd like to have solved

The built-in asyncify is very useful, but the way it's implemented means that typing checking (such as via Pyre) and IDEs have no idea these exist.

### Describe the ideal solution

Add function stubs for every method that supports async with type hinting.

**Comment by Avasam** [comment]
https://github.com/auth0/auth0-python/pull/678#issuecomment-2798399228
> Here's one way you could solve this (without changing anything at runtime): Generate stubs for all classes you expect to "Asyncify" as was talked here. But with the following differences:
> 
> * Generate it in a subfolder that doesn't exist at runtime (ie: `auth0._asyncified.*`)
> * Rename all the generated classes to help differentiate them (something like `class UsersWrapped(Users):` for base class `Users`)
> * Decorate all generated classes with `@typing.type_check_only`
> * Generate a massive overload for `auth0.asyncify.asyncify` that maps original classes to stub-only asyncified/wrapped classes.
> * Optional: Generate said stubs on install/editable install/wheel creation. That way they don't pollute the source and history.

> Like this: [Avasam/typeshed#43 (files)](https://github.com/Avasam/typeshed/pull/43/files#diff-2e519b89160890743c6aaa1d6e57c21160a2db3148ce891a8e5aa250b6bc9b62)

**Comment by kishore7snehil** [comment]
I believe this issue can be closed now as we already have a workaround discussed on this thread.

**Comment by kishore7snehil** [comment]
@Dreamsorcerer There was a PR for the same  where it was asked to add the `.py` files in the repo itself. I suggested to use the [script](https://github.com/auth0/auth0-python/pull/678#issuecomment-2737933954) to generate the async files.

That's the same approach @Avasam (correct me if I am wrong ) is pushing forward but with some specific tweaks to the generation logic.

If he is willing to add a PR to this repo, I can review it and take it forward.

**Comment by Dreamsorcerer** [comment]
> If he is willing to add a PR to this repo, I can review it and take it forward.

Of course. My point was just that this issue isn't closed until that PR gets created and merged.

**Comment by kishore7snehil** [comment]
@Avasam  Please feel free to open a PR. I can review it and will update you acordingly.
